pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+          sharedBeforeEach('grants the permissions', async () => {
+            await authorizer.grantPermissionGlobally(ACTION_1, user, { from: root });
+          });
+          context('when revoking the permission for a contract', () => {
+            it('still can perform the requested action for the requested contract', async () => {
+              await authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_1)).to.be.true;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.true;
+            });
+
+            it('still can perform the requested action everywhere', async () => {
+              await authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.true;
+            });
+
+            it('does not emit an event', async () => {
+              const tx = await authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: getSender() });
+              expectEvent.notEmitted(await tx.wait(), 'PermissionRevoked');
+            });
+          });
+          context('when revoking the permission for a everywhere', () => {
+            it('revokes the requested global permission and cannot perform the requested action everywhere', async () => {
+              await authorizer.revokePermissionGlobally(ACTION_1, user, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_1)).to.be.false;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.false;
+            });
+
+            it('cannot perform the requested action in any specific contract', async () => {
+              await authorizer.revokePermissionGlobally(ACTION_1, user, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, NOT_WHERE)).to.be.false;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.false;
+            });
+
+            it('emits an event', async () => {
+              const receipt = await (
+                await authorizer.revokePermissionGlobally(ACTION_1, user, { from: getSender() })
+              ).wait();
+
+              expectEvent.inReceipt(receipt, 'PermissionRevoked', {
+                actionId: ACTION_1,
+                account: user.address,
+                where: TimelockAuthorizer.EVERYWHERE,
+              });
+            });
+          });
+        });
+      }
+
+      context('when the sender is root', () => {
+        itRevokesPermissionCorrectly(() => root);
+      });
+
+      context('when the sender is revoker', () => {
+        sharedBeforeEach('makes a revoker', async () => {
+          await authorizer.addRevoker(revoker, EVERYWHERE, { from: root });
+        });
+        itRevokesPermissionCorrectly(() => revoker);
+      });
+    });
```

Comment:
Same suggestions regarding permission checks for this and scheduleRevoke as I submitted for grant and scheduleGrant (since they're all similar)